src/SFA.DAS.Employer.PR.Web/Controllers/SelectTrainingProviderController.cs

```diff
@@ -12,26 +13,28 @@ namespace SFA.DAS.Employer.PR.Web.Controllers;
 
 [Authorize(Policy = nameof(PolicyNames.HasEmployerOwnerAccount))]
 [Route("accounts/{employerAccountId}/providers/new/selectProvider", Name = RouteNames.SelectTrainingProvider)]
-public class SelectTrainingProviderController(ISessionService _sessionService, IValidator<SelectTrainingProviderSubmitViewModel> _validator) : Controller
+public class SelectTrainingProviderController(IOuterApiClient _outerApiClient, ISessionService _sessionService, IValidator<SelectTrainingProviderSubmitModel> _validator) : Controller
 {
+    public const string ShutterPageViewPath = "~/Views/SetPermissions/ShutterPage.cshtml";
+
     [HttpGet]
     public IActionResult Index([FromRoute] string employerAccountId)
     {
         var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
 
-        if (sessionModel == null)
+        if (sessionModel?.LegalEntityId == null || sessionModel.EmployerAccountId != employerAccountId)
         {
             return RedirectToAction("Index", "YourTrainingProviders", new { employerAccountId });
         }
 
         var backLink = SetBackLink(employerAccountId, sessionModel.AccountLegalEntities.Count);
 
-        SelectTrainingProviderViewModel model = new SelectTrainingProviderViewModel(backLink!, sessionModel!.ProviderName, sessionModel!.Ukprn.ToString());
+        SelectTrainingProviderModel model = new SelectTrainingProviderModel(backLink!, sessionModel!.ProviderName, sessionModel!.Ukprn.ToString());
         return View(model);
     }
 
     [HttpPost]
-    public IActionResult Index([FromRoute] string employerAccountId, SelectTrainingProviderSubmitViewModel submitModel)
+    public async Task<IActionResult> Index([FromRoute] string employerAccountId, SelectTrainingProviderSubmitModel submitModel, CancellationToken cancellationToken)
     {
         var sessionModel = _sessionService.Get<AddTrainingProvidersSessionModel>();
```

Comment:
SessionModel can be null if the session has timed out, we should always check if the model exists.

Reply:
ok